### ISSUE TITLE
Accuracy, Precision, Recall, Specificity wrong accumulation with average='none' inside Wrapper.

### ISSUE THREAD

**Issue by kpoeppel** [PyTorchLightning/metrics#1045]
## 🐛 Bug

The Classification Metrics Accuracy, Precision, Recall and Specificity in average='none' mode return negative metrics values, when called multiple times once with missing and once with present targets. 

### To Reproduce

```py
import torchmetrics as tm
import torch


class MetricWrapper(tm.Metric):
    def __init__(self, metric):
        super().__init__()
        self.metric = metric
    
    def update(self, *args, **kwargs):
        self.metric.update(*args, **kwargs)
    
    def compute(self, *args, **kwargs):
        return self.metric.compute(*args, **kwargs)



def main():
    acc = MetricWrapper(tm.Accuracy(num_classes=3, average='none'))
    # EDIT!
    # pred = torch.tensor([[0., 1., 0.], [0., 0., 1.]])
    # to
    pred = torch.tensor([[0., 1., 0.], [1., 0., 0.]])
    target = torch.tensor([0, 1])

    pred2 = torch.tensor([[0., 1., 0.], [1., 0., 0.]])
    target2 = torch.tensor([0, 2])

    print(acc(pred, target))

    print(acc(pred2, target2))


if __name__ == "__main__":
    main()
```

Output:
tensor([0., 0., nan])
tensor([0., 0., -1.])

### Expected behavior

Output:
tensor([0., 0., nan])
tensor([0., 0., 0.])

### Environment

- TorchMetrics version: 0.8.2

### Additional context
The origin of the problem is already found: Bad handling of mutability.
PR follows.
